### ISSUE TITLE
ability set dynamic headers per request

### ISSUE THREAD

**Issue by Fivell** [swisnl/json-api-client#33]
## Detailed description
From the source code it is not clear how to attach dynamic header for each request

## Context
Different users should have ability to use they own  API KEY  to perform requests

**Comment by JaZo** [comment]
Hi @Fivell, I have added a headers param to the request methods of `DocumentClient` (#34). This allows you to use the `DocumentClient` and set (authorization) headers per request.

**Comment by Fivell** [comment]
@JaZo  thanks, also we found a way to implement it with custom client class, like
```
class ClientWithApiKey extends \Swis\JsonApi\Client\Client
{
  public function setApiKey(string $apiKey)
  {
    $this->defaultHeaders = $this->mergeHeaders(['api-key' => $apiKey]);
  }
}
```

**Comment by Fivell** [comment]
@JaZo  can u give an example how u  see this can be implemented after #34  ?

**Comment by JaZo** [comment]
@Fivell, this can be used like the following:

```php
$client = app(DocumentClientInterface::class);
$document = $client->get('blogs', ['api-key' => 'foo-bar']);
```
